File: shared/src/commonMain/kotlin/com/tom/paperless/domain/models/uiStates/NfcTaggablesUiState.kt
package com.tom.paperless.domain.models.uiStates

import com.tom.paperless.domain.models.NfcTaggable
import com.tom.paperless.domain.models.enums.TargetType

data class NfcTaggablesUiState(
    val items: List<NfcTaggable> = emptyList(),
    val isLoading: Boolean = false,
    val errorMessage: String? = null,

    val selectedItem: NfcTaggable? = null,
    val activeTypeFilter: TargetType? = null,
    val searchQueryText: String = ""
)

File: iosApp/PaperLess/Application/Ultility/Extensions/NfcTaggablesUiState.swift
//
//  NfcTaggablesUiState.swift
//  PaperLess
//
//  Created by Tom Salih on 23.09.25.
//

import Shared

extension NfcTaggablesUiState {
    static func empty() -> NfcTaggablesUiState {
        NfcTaggablesUiState(
            items: [],
            isLoading: false,
            errorMessage: nil,
            selectedItem: nil,
            activeTypeFilter: nil,
            searchQueryText: ""
        )
    }
}
